Automattic/woocommerce-payments
# Add Affirm and Afterpay to Block Checkout

Fixes #6231 

#### Changes proposed in this Pull Request

* Provide option of Affirm and Afterpay at checkout for eligible merchants on a block-based checkout page

#### Testing instructions

##### Local server preparation
* Check in to any test account of an eligible country. Eg. US - `acct_1LqHU4QocNIKfAAZ` . 
* Add dummy BNPL related fees to the local setup with these commands 
```
wp wcpay add_new_fee base base-us-affirm-fee usd --percentage_rate=0.042 --fixed_rate=42 --countries=US --payment_methods=affirm --mode=test --description="Affirm test"
wp wcpay add_new_fee base base-us-afterpay-fee usd --percentage_rate=0.042 --fixed_rate=42 --countries=US --payment_methods=afterpay_clearpay --mode=test --description="Afterpay test"
```
Add these for any country that you wish to test with by modifying the command accordingly 
* Set the BNPL flag to true
`wp wcpay set_account_meta acct_1LqHU4QocNIKfAAZ --key=bnpl_affirm_afterpay_enabled --value=1`

Make sure you use your test account number in the command.

##### Test
* Browse to `Payments > Settings` in WP Admin and make sure you turn on Affirm and Afterpay. 
* If not done already - create a block based checkout page, and set it as the checkout page from `WooCommerce > Settings > Advanced tab > Page setup` 
* Add items to the cart with total amount within the required threshold for the method you are testing ( this check is being addressed in a [separate PR](https://github.com/Automattic/woocommerce-payments/pull/6540) ) . 
    *  Affirm threshold - [For US](https://stripe.com/docs/payments/affirm#payment-plans-us) | [For Canada](https://stripe.com/docs/payments/affirm#payment-plans-ca)
    * [Afterpay thresholds](https://stripe.com/docs/payments/afterpay-clearpay#collection-schedule)
* Watch for any errors in console or in the front-end flow. 

Use the WCPay Dev tools and test the scenarios above for both UPE Classic and Split. 

-------------------


**Post merge**
